pkg/aws/policies.go

```diff
@@ -272,21 +272,29 @@ func (c *awsClient) EnsurePolicy(policyArn string, document string,
 	}
 
 	if !isCompatible {
+		// Since there is a limit to how many versions a policy can have, we delete all non-default
+		// policy versions from the list, thus making space for the new one.
+		policyVersionsOutput, err := c.iamClient.ListPolicyVersions(&iam.ListPolicyVersionsInput{
+			PolicyArn: aws.String(policyArn),
+		})
+		if err != nil {
+			return policyArn, err
+		}
+		for _, version := range policyVersionsOutput.Versions {
+			if aws.BoolValue(version.IsDefaultVersion) {
+				continue
+			}
+			c.iamClient.DeletePolicyVersion(&iam.DeletePolicyVersionInput{
```

Comment:
@vkareh  Do you think we can leave the latest one before deleting it? Because if the upgrade fails or create policy version fails, we risk deleting everything. wdyt?

Reply:
The latest one (default) will remain, so after this function runs you will always have the current/default and the previous one.